{{cookiecutter.repo_name}}/setup.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# pylint: disable=C0111,W6005,W6100
+from __future__ import absolute_import, print_function
+
+import os
+import re
+import sys
+
+from setuptools import setup
+
+{%- set license_classifiers = ['AGPL 3.0', 'Apache Software License 2.0'] %}
+
+
+def get_version(*file_paths):
+    """Extract the version string from the file at the given relative path fragments."""
+    filename = os.path.join(os.path.dirname(__file__), *file_paths)
+    version_file = open(filename).read()
+    version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]",
+                              version_file, re.M)
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError('Unable to find version string.')
+
+VERSION = get_version('{{ cookiecutter.app_name }}', '__init__.py')
+
+if sys.argv[-1] == 'tag':
+    print("Tagging the version on github:")
+    os.system("git tag -a %s -m 'version %s'" % (VERSION, VERSION))
+    os.system("git push --tags")
+    sys.exit()
+
+README = open(os.path.join(os.path.dirname(__file__), 'README.rst')).read()
+CHANGELOG = open(os.path.join(os.path.dirname(__file__), 'CHANGELOG.rst')).read()
+
+setup(
+    name='{{ cookiecutter.repo_name }}',
+    version=VERSION,
+    description="""{{ cookiecutter.project_short_description }}""",
+    long_description=README + '\n\n' + CHANGELOG,
+    author='edX',
+    author_email='oscm@edx.org',
+    url='https://github.com/edx/{{ cookiecutter.repo_name }}',
+    packages=[
+        '{{ cookiecutter.app_name }}',
+    ],
+    include_package_data=True,
+    install_requires=[{% if cookiecutter.models != "Comma-separated list of models" %}"django-model-utils>=2.0"{% endif %}],
```

Comment:
`install_requires` is for minimum requirements, requirements files are for specifying what to install in particular environments; here's a [writeup of the distinction](https://packaging.python.org/requirements/#install-requires-vs-requirements-files) from the Python Packaging User Guide.  The way they're intended to be used in projects generated from this template is:

* Core requirements are listed here in `install_requires`, and only modified when code changes are made which require a new package or drop support for a previously supported one
* Direct dependencies are listed in `requirements/*.in` for use in development and tox virtualenvs with as few version constraints as possible
* Full pinned dependencies in `requirements/*.txt` are auto-generated via `make pip-compile`

I definitely want to document decisions like this, but the cookiecutter seems like the wrong place for that documentation; really belongs in OEPs instead.